src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -71,67 +71,77 @@ public async Task StartAssociatedReleasePipelinesAsync(int buildId, int channelI
+                            {
+                                Logger.LogError($"Could not find the specified BAR Build {item.BuildId} to run a release pipeline.");
+                            }
+                            else if (build.AzureDevOpsBuildId == null)
                             {
-                                await HandleCompletedBuild(item, azdoBuild, cancellationToken);
+                                // If something uses the old API version we won't have this information available.
+                                // This will also be the case if something adds an existing build (created using
+                                // the old API version) to a channel
+                                Logger.LogInformation($"barBuildInfo.AzureDevOpsBuildId is null for BAR Build.Id {build.Id}.");
                             }
                             else
                             {
-                                Logger.LogInformation($"AzDO build {azdoBuild.BuildNumber}/{azdoBuild.Definition.Name} with BAR BuildId {build.Id} is still in progress.");
+                                AzureDevOpsClient azdoClient = await GetAzureDevOpsClientForAccount(build.AzureDevOpsAccount);
+
+                                var azdoBuild = await azdoClient.GetBuildAsync(
+                                    build.AzureDevOpsAccount,
+                                    build.AzureDevOpsProject,
+                                    build.AzureDevOpsBuildId.Value);
 
-                                // Build didn't finish yet. Let's wait some time and try again.
-                                EnqueueBuildStatusCheck(item, 0);
+                                if (azdoBuild.Status.Equals("completed", StringComparison.OrdinalIgnoreCase))
+                                {
+                                    await HandleCompletedBuild(item, azdoBuild, cancellationToken);
+                                }
+                                else
+                                {
+                                    Logger.LogInformation($"AzDO build {azdoBuild.BuildNumber}/{azdoBuild.Definition.Name} with BAR BuildId {build.Id} is still in progress.");
+
+                                    // Build didn't finish yet. Let's wait some time and try again.
+                                    EnqueueBuildStatusCheck(item, 0);
+                                }
                             }
                         }
-                    }
 
-                    await tx.CommitAsync();
+                        await tx.CommitAsync();
+                    }
+                }
+                catch (TaskCanceledException tcex) when (tcex.CancellationToken == cancellationToken)
+                {
+                    return TimeSpan.MaxValue;
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError(ex, "Processing queue messages");
+                    break;
```

Comment:
Won't be better to let the runner go to next queue item?

Reply:
In this case, because the transaction wasn't committed, you'll just end up dequeing the same item again and end up in an infinite loop that runs without any delay between iterations.